Source/UI/Coordinators/PasswordCoordinator.swift

```diff
@@ -139,12 +139,19 @@ extension PasswordCoordinator {
             switch result {
             case let .success(currentUser):
                 self?.configuration.tracker?.loginID = currentUser.legacyID
-                self?.updatekeyChain(
-                    for: identifier,
-                    loginFlowVariant: loginFlowVariant,
-                    password: password
-                ) {
-                    self?.spawnCompleteProfileCoordinator(for: .signin(user: currentUser), persistUser: persistUser, completion: completion)
+                let device = UserDevice(
+                    hash: nil,
+                    applicationName: self?.configuration.appName,
+                    applicationVersion: self?.configuration.appVersion
+                )
+                currentUser.device.update(device) { _ in
+                    self?.updatekeyChain(
```

Comment:
Does this need to be nested as a callback? If possible I think it would make sense to move the `self?.updatekeyChain` out of the callback to avoid waiting on the device fingerprint request to be completed (since we don't really care about the result).